src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -30,15 +30,14 @@ ImmutabilityContext context
 
 			var members = type.GetMembers()
 				// Exclude static members.
-				// Immutability is, for us, a property held by values. A type
-				// can be immutable in one of two ways and these are statements
-				// about values of that type.
-				// Static members are not tied to values/instances by
-				// definition. Although we care about them otherwise, TODO explain
+				// Immutability is a property of values, [Immutable] (etc.) are judgements
+				// about the immutability of values of some type.
+				// Static fields/properties are global variables scoped to particular types,
+				// but are not a factor in the immutability of the values of that type themselves.
 				.Where( m => !m.IsStatic );
 
-			// Check that the base class is immutable for classes
 			if( type.TypeKind == TypeKind.Class ) {
+				// Check that the base class is immutable for classes

```

Comment:
Most of the code has the comments inside the if blocks 🤷 this PR standardizes on that.